include/boost/gil/image_processing/morphology.hpp

```diff
@@ -0,0 +1,282 @@
+  gil_function_requires<MutableImageViewConcept<DstView>>();
+
+  gil_function_requires<ColorSpacesCompatibleConcept<
+                              typename color_space_type<SrcView>::type,
+                              typename color_space_type<DstView>::type>>();
+
+  gil::image<typename DstView::value_type> intermediate_img(src_view.dimensions());
+
+  for (std::size_t i = 0; i < src_view.num_channels(); i++) {
+    morph_impl(nth_channel_view(src_view, i),
+               nth_channel_view(view(intermediate_img), i), ker_mat,
+               identifier);
+  }
+  copy_pixels(view(intermediate_img), dst_view);
+}
+
+/// \brief Calculates the difference between pixel values of first image_view
+/// and second image_view. \param src_view1 - First parameter for subtraction of
+/// views. \param src_view2 - Second parameter for subtraction of views. \param
+/// diff_view - View containing result of the subtraction of second view from
+/// the first view. \tparam SrcView type of source/Input images used for
+/// subtraction. \tparam DiffView type of image view containing the result of
+/// subtraction.
+template <typename SrcView, typename DiffView>
+void difference_impl(SrcView const &src_view1, SrcView const &src_view2,
+                     DiffView const &diff_view) {
+  for (std::ptrdiff_t view_row = 0; view_row < src_view1.height(); ++view_row)
+    for (std::ptrdiff_t view_col = 0; view_col < src_view1.width(); ++view_col)
+      diff_view(view_col, view_row) =
+          src_view1(view_col, view_row) - src_view2(view_col, view_row);
+}
+
+/// \brief Passes parameter values to the function 'difference_impl' alongwith
+/// individual channel views of input images. \param src_view1 - First parameter
+/// for subtraction of views. \param src_view2 - Second parameter for
+/// subtraction of views. \param diff_view - View containing result of the
+/// subtraction of second view from the first view. \tparam SrcView type of
+/// source/Input images used for subtraction. \tparam DiffView type of image
+/// view containing the result of subtraction.
+template <typename SrcView, typename DiffView>
+void difference(SrcView const &src_view1, SrcView const &src_view2,
+                DiffView const &diff_view) {
+  gil_function_requires<ImageViewConcept<SrcView>>();
+  gil_function_requires<MutableImageViewConcept<DiffView>>();
+
+  gil_function_requires<ColorSpacesCompatibleConcept<
+                              typename color_space_type<SrcView>::type,
+                              typename color_space_type<DiffView>::type>>();
+
+  for (std::size_t i = 0; i < src_view1.num_channels(); i++) {
+    difference_impl(nth_channel_view(src_view1, i),
+                    nth_channel_view(src_view2, i),
+                    nth_channel_view(diff_view, i));
+  }
+}
+} // namespace detail
+
+// IntOpView : View utilized for performing intermediate operations which will
+// contribute in creating the resultant view .
```

Comment:
Leftover?

Reply:
I thought that it would be good to express what I had in my mind while naming IntOpView to the reviewers, it was not leftover.

Reply:
Not sure I'm following. Perhaps it would be better to add it in the documentation of the functions then? You could also name it like `InterimOpView` which is a bit more obvious. Something along the lines of:

```
\param int_op_view - view for writing output and performing intermediate operations
\tparam IntOpView type of output image, models gil::MutableImageViewConcept

```

Reply:
For now, I have provided parameter definitions in the same way as you suggested. Do tell me if I need to tweak parameter name too .